rq/queue.py

```diff
@@ -329,42 +410,46 @@ def enqueue_call(self, func, args=None, kwargs=None, timeout=None,
             retry=retry
         )
 
-        # If a _dependent_ job depends on any unfinished job, register all the
-        # _dependent_ job's dependencies instead of enqueueing it.
-        #
-        # `Job#fetch_dependencies` sets WATCH on all dependencies. If
-        # WatchError is raised in the when the pipeline is executed, that means
-        # something else has modified either the set of dependencies or the
-        # status of one of them. In this case, we simply retry.
-        if depends_on is not None:
-            with self.connection.pipeline() as pipe:
-                while True:
-                    try:
-
-                        pipe.watch(job.dependencies_key)
-
-                        dependencies = job.fetch_dependencies(
-                            watch=True,
-                            pipeline=pipe
-                        )
-
-                        pipe.multi()
-
-                        for dependency in dependencies:
-                            if dependency.get_status(refresh=False) != JobStatus.FINISHED:
-                                job.set_status(JobStatus.DEFERRED, pipeline=pipe)
-                                job.register_dependency(pipeline=pipe)
-                                job.save(pipeline=pipe)
-                                job.cleanup(ttl=job.ttl, pipeline=pipe)
-                                pipe.execute()
-                                return job
-
-                        break
-                    except WatchError:
-                        continue
+        return self.setup_dependencies_and_enqueue(
+            job,
+            pipeline=pipeline,
+            at_front=at_front
+        )
 
-        job = self.enqueue_job(job, at_front=at_front)
-        return job
+    def enqueue_many(
+        self,
+        job_datas,
+        pipeline=None
+    ):
+        pipe = pipeline if pipeline is not None else self.connection.pipeline()
+        while True:
+            try:
+                pipe.multi()
+                jobs = [
+                    self.enqueue_job(
+                        self.create_job(
+                            job_data.func, args=job_data.args, kwargs=job_data.kwargs, result_ttl=job_data.result_ttl,
+                            ttl=job_data.ttl,
+                            failure_ttl=job_data.failure_ttl, description=job_data.description,
+                            depends_on=None,
+                            job_id=job_data.job_id, meta=job_data.meta, status=JobStatus.QUEUED,
+                            timeout=job_data.timeout,
+                            retry=job_data.retry
+                        ),
+                        pipeline=pipe,
+                        at_front=job_data.at_front
+                    )
+                    for job_data in job_datas
+                ]
+                if pipeline is None:
+                    pipe.execute()
+                return jobs
+            except WatchError:
```

Comment:
Do we need to catch `WatchError` here if `depends_on` is always `None`?

Reply:
I guess not, if `pipeline` is passed in without the `except` it would be raised to them, which is what we are doing here anyway.  I can drop this code.